### DISCUSSION TITLE
[v5]how to orderBy preload column ?

### DISCUSSION THREAD

**Discussion by BlazingCode** [adonisjs/core#4100] [Help] [answered]
there is User , Document Model

I want it to work with

`Document.query().preload('user').orderBy('user.name','desc') `

i know if use join
```
const documents= await Document
  .query()
  .preload('user')
  .join('users', 'users.id', 'document.user_id')
  .orderBy('users.name', 'desc')
  .select('document.*')
```

Lazy Loading method load() need findOrFail || find function so i can`t use it 

Is there any way without using join or orderByRaw?

or  how can use lazy loding way?

**Comment by sergefabre** [comment]
Hello
are you test 
`Document.query().preload('user', (user) => user.orderBy(name,'desc'))` ?

**Reply by BlazingCode** [reply]
thanks for the answer

I have already tested the query you suggested.

what i want
Sort the documents based on the columns of the user model.

The relationship between user and document is a 1:N relationship

The above query is not suitable for my situation

**Reply by sergefabre** [reply]
in your model, you use the relation hasMany ?

**Reply by sergefabre** [reply]
`Document.query().preload('user', (u) => u.orderBy('name','desc'))`

**Comment by RomainLanz** [comment, marked as answer]
Hey @BlazingCode! 👋🏻 

I recommend writing the query in SQL first without relying on the ORM.
Once you have a query that meets your requirements, we can consider how to integrate it into the ORM syntax.
